theano/compile/function_module.py

```diff
@@ -676,6 +676,18 @@ def __call__(self, *args, **kwargs):
         elif self.unpack_single and len(outputs) == 1:
             return outputs[0]
         else:
+
+            if self.output_keys is not None: 
+                outputDict = {}
+
+
+                assert len(self.output_keys) == len(outputs)
+
+                #for i in range(0, len(self.output_keys)): 
+                #    outputDict[self.output_keys[i]] = outputs[i]
+
+                return dict(itertools.izip(self.output_keys, outputs))
```

Comment:
They're equally fast, so I went with the shorter one.  In either case, it's unlikely to be a bottleneck in function calls unless one is using a very large number of outputs, which is a bad idea anyway.